src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1803,16 +1732,13 @@ void CodeGen::genSSE41Intrinsic(GenTreeHWIntrinsic* node)
     regNumber op4Reg = REG_NA;
     emitter*  emit   = getEmitter();
 
-    if ((op1 != nullptr) && !op1->OperIsList())
-    {
-        op1Reg = op1->gtRegNum;
-        genConsumeOperands(node);
-    }
+    genConsumeHWIntrinsicOperands(node);
```

Comment:
Shouldn't there be some additional removals of `genConsumeReg` from the below (for this and similar edits)?

Reply:
I didn't transform all the methods in this file to use `genConsumeHWIntrinsicOperands` - only the ones where it was specifically beneficial to abstract out the possibility that they might be contained memory ops/addresses. That may be a useful future change, but I wouldn't necessarily do that with this PR.